### PR TITLE
Add winflexbison v2.5.20

### PR DESCRIPTION
**Pull request by seanyen**
Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by scopatz** [comment]
Seems like this is failing on windows

**Comment by seanyen** [comment]
cc @conda-forge/staged-recipes this is ready for review and merge.

Btw, for the `appveyor` failure is little mysterious to me. I kicked off the same build but it came out with different results:
* Passed one: https://ci.appveyor.com/project/conda-forge/staged-recipes/builds/29152967
* Failed one: https://ci.appveyor.com/project/conda-forge/staged-recipes/builds/29169771

It seemed on the failed one, the machine doesn't have Vs2017 installed?

**Comment by isuruf** [comment]
Why do you need VS2017 in the first place?

**Comment by seanyen** [comment]
> Why do you need VS2017 in the first place?

I don't have strong reason to be in favor of VS2017 but it comes with better language standards support and many other improvements, so it is straightforward for me to choose that one. Is there any considerations not to encourage doing so?

**Comment by isuruf** [comment]
>  Is there any considerations not to encourage doing so?

Yes, it's better to go with the default compilers and upgrade when all packages upgrade. There might be ABI differences when compiled with VS2015 vs VS2017 (With -GL turned on).

File: recipes/winflexbison/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+source:
+  - url: https://github.com/lexxmark/{{ name }}/archive/v{{ version }}.tar.gz
+    sha256: cd29f888f167128c22211444c6e31b5f4823a7de0e23d8039559ee87b80349c9
+    patches:
+      - fix-install.patch
+
+build:
+  number: 0
+  skip: true  # [not win or py2]
```
Comment (isuruf) [inline recipes/winflexbison/meta.yaml]:
```suggestion
  skip: true  # [not win or vc<14]
```

Comment (seanyen) [reply recipes/winflexbison/meta.yaml]: Thanks for the tips. The `vc` selector can be used in this case.

**Comment by seanyen** [comment]
@isuruf It looks all green now. Let me know if anything needs touch-ups. Thanks.

**Comment by seanyen** [comment]
cc @conda-forge/staged-recipes this is ready for review and merge. thanks!

File: recipes/winflexbison/bld.bat [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    -G "Ninja" ^
+    -DCMAKE_INSTALL_PREFIX=%LIBRARY_PREFIX% ^
+    -DCMAKE_BUILD_TYPE=Release ^
+    -DCMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP=True ^
+    -DBUILD_SHARED_LIBS=ON ^
+    %SRC_DIR%
+if errorlevel 1 exit 1
+
+:: Build.
+cmake --build . --config Release
```
Comment (isuruf) [inline recipes/winflexbison/bld.bat]: Can you make this use `%CPU_COUNT%`?

Comment (isuruf) [reply recipes/winflexbison/bld.bat]: `ninja -j%CPU_COUNT%` should work.

Comment (seanyen) [reply recipes/winflexbison/bld.bat]: Thanks. I added the `%CPU_COUNT%`. JFYI, I also found CMake seems to consume a environment variable [`CMAKE_BUILD_PARALLEL_LEVEL`](https://cmake.org/cmake/help/v3.12/envvar/CMAKE_BUILD_PARALLEL_LEVEL.html#envvar:CMAKE_BUILD_PARALLEL_LEVEL) and perhaps `conda-forge-ci-setup` could set it by default in future? What do you think?

Comment (isuruf) [reply recipes/winflexbison/bld.bat]: Thanks for the info. Yes, that would be great to have. PRs are welcome to conda-forge-ci-setup-feedstock.

**Pull request merged**